### PR TITLE
feat(atenlib): add op(chunk)

### PR DESCRIPTION
**Pull request by xiaowuhu**

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
LGTM! I would wait for Rama to review the parts in the evaluator

**Comment by xiaowuhu** [comment]
> LGTM! I would wait for Rama to review the parts in the evaluator

@gramalingam pls help to review. The original implementation doesn't return valid output numbers and based on opset17(split).

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -955,10 +955,30 @@ def aten_choose_qparams_optimized(
...
-    raise NotImplementedError()
+    neg_1 = op.Constant(value_ints=[-1])
+    # get size of specified dim
+    self_shape = op.Shape(self)
+    dim_size = op.Gather(self_shape, dim, axis=0)
+    # cal size/chunk, get number of data in one chunk
+    num_per_chunk = op.Div(dim_size, chunks)
+    if op.Mod(dim_size, chunks) > 0:  # type: ignore[operator]
+        # cannot be divisible, then num_in_chunk + 1
+        num_per_chunk = op.Add(num_per_chunk, 1)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:970]: I wonder if using Where will be more efficient or not

Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:970]: Even better: `op.Cast(op.Mod(dim_size, chunks) > 0, to=INT64.value) + num_per_chunk`

Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:970]: The take away for me was `If` tends to be slow and should be replaced with other things when possible.

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 2]
```diff
@@ -955,10 +955,30 @@ def aten_choose_qparams_optimized(
...
+
+    # cal real chunk number
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:972]:
```suggestion
    # calculate real chunk number
```

typo?

Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:972]: Trying to understand: why was the above not "real"?

**Comment by justinchuby** [comment]
Also cc @BowenBao for the logic

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 2 above)
Comment (xiaowuhu) [reply onnxscript/function_libs/torch_aten/ops/core.py:972]: I think it can also work on BOOL value

**Comment by justinchuby** [comment]
We can merge if there's no movement

**Pull request merged**